### PR TITLE
[CARBONDATA-4053] Fix alter table rename column failed when column name is "a"

### PR DESCRIPTION
**Pull request by jack86596**
### Why is this PR needed?
Alter table rename column failed because incorrectly replace the content in tblproperties by new column name, which the content is not related to column name.
 
 ### What changes were proposed in this PR?
Instead of calling replace method on property value directly, first filter out the properties which related to column name, then find the matched old column name, replace it with new name.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

File: integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala [hunk 1]
```diff
@@ -332,12 +332,26 @@ object AlterTableUtil {
       tableProperties: mutable.Map[String, String],
       oldColumnName: String,
       newColumnName: String): Unit = {
+    val columnProperties = Seq("NO_INVERTED_INDEX",
+      "INVERTED_INDEX",
+      "INDEX_COLUMNS",
+      "COLUMN_META_CACHE",
+      "DICTIONARY_INCLUDE",
```
Comment (ajantha-bhat) [inline integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala:339]: we have removed  DICTIONARY_INCLUDE, DICTIONARY_EXCLUDE from 2.0, please check the applicable table properties and keep only those also add if something new added in 2.0

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala [hunk 2]
```diff
@@ -38,6 +38,15 @@ class AlterTableColumnRenameTestCase extends QueryTest with BeforeAndAfterAll {
     assert(null == carbonTable.getColumnByName("empname"))
   }
 
+  test("CARBONDATA-4053 test rename column when column name is a") {
+    sql("create table simple_table(a int) stored as carbondata")
+    sql("alter table simple_table change a a1 int")
+    val carbonTable = CarbonMetadata.getInstance().getCarbonTable("default", "simple_table")
```
Comment (ajantha-bhat) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala:44]: This issue happens when you have table properties say SORT_COLUMNS = "a,b"
Then the column name changed from a to a1, then it was changing SORT_COLUMNS = "a1" instead of SORT_COLUMNS = "a1,b". So, your current testcase cannot reproduce issue. so, add the testcase that gives issue without the change by keeping some table properties.

File: integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala (hunk 1 above)
Comment (ajantha-bhat) [reply integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala:339]: Also INDEX_COLUMNS are changed to SPATIAL_INDEX I think, refer docs/spatial-index-guide.md

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4870/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3117/

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala (hunk 2 above)
Comment (jack86596) [reply integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala:45]: Done, modify the testcase, now can cover the scenario you mentioned.

File: integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala (hunk 1 above)
Comment (jack86596) [reply integration/spark/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: Removed "DICTIONARY_INCLUDE" and "DICTIONARY_EXCLUDE". All properties which related to column name are now included in the list. 
"INDEX_COLUMNS" and "SPATIAL_INDEX" is the property only for bloom index and spatial index table, which these two kinds of table are blocked to rename the column, so it is ok for these two properties not to be included in the property list.

**Comment by ajantha-bhat** [comment]
LGTM.
Thanks for your contribution.

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala [hunk 3]
```diff
@@ -98,7 +108,7 @@ class AlterTableColumnRenameTestCase extends QueryTest with BeforeAndAfterAll {
     // Bucket Column
     sql("DROP TABLE IF EXISTS rename_bucket")
     sql("CREATE TABLE rename_bucket (ID Int, date Timestamp, country String, name String)" +
-      " STORED AS carbondata TBLPROPERTIES ('BUCKET_NUMBER'='4', 'BUCKET_COLUMNS'='name')")
+        " STORED AS carbondata TBLPROPERTIES ('BUCKET_NUMBER'='4', 'BUCKET_COLUMNS'='name')")
```
Comment (marchpure) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala:111]: please revert this change of format

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala [hunk 4]
```diff
@@ -81,7 +91,7 @@ class AlterTableColumnRenameTestCase extends QueryTest with BeforeAndAfterAll {
     // Non-Partition Column with Complex Datatype
     sql("DROP TABLE IF EXISTS rename_complextype")
     sql(s"create table rename_complextype(mapcol map<string,string>," +
-      s" arraycol array<string>) stored as carbondata")
+        s" arraycol array<string>) stored as carbondata")
```
Comment (marchpure) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala:94]: please revert this change of format

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala [hunk 5]
```diff
@@ -472,10 +483,10 @@ class AlterTableColumnRenameTestCase extends QueryTest with BeforeAndAfterAll {
   def createPartitionTableAndLoad(): Unit = {
     sql(
       "CREATE TABLE rename_partition (empno int, empname String, designation String," +
-        " doj Timestamp, workgroupcategory int, workgroupcategoryname String, deptno int," +
-        " deptname String," +
-        " projectjoindate Timestamp, projectenddate Timestamp,attendance int," +
-        " utilization int,salary int) PARTITIONED BY (projectcode int) STORED AS carbondata")
+      " doj Timestamp, workgroupcategory int, workgroupcategoryname String, deptno int," +
```
Comment (marchpure) [inline integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala:486]: please revert this change of format

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3128/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4882/

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala (hunk 3 above)
Comment (jack86596) [reply integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala]: Done.

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala (hunk 4 above)
Comment (jack86596) [reply integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala]: Done.

File: integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala (hunk 5 above)
Comment (jack86596) [reply integration/spark/src/test/scala/org/apache/spark/carbondata/restructure/vectorreader/AlterTableColumnRenameTestCase.scala]: Done.

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4885/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3131/
